src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs

```diff
@@ -131,6 +131,7 @@ public partial interface IImmutableDictionary<TKey, TValue> : System.Collections
         System.Collections.Immutable.IImmutableDictionary<TKey, TValue> SetItems(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<TKey, TValue>> items);
         bool TryGetKey(TKey equalKey, out TKey actualKey);
     }
+    [System.Runtime.CompilerServices.CollectionBuilderAttribute(typeof(System.Collections.Immutable.ImmutableList), "Create")]
```

Comment:
Presumably the compiler will use this even if the return type is a subtype of the interface?

Reply:
That's the discussion here:
https://github.com/dotnet/runtime/pull/89459#issuecomment-1650365154

We'll merge this iff Roslyn confirms they need this and it'll be supported. Otherwise, we'll close it.

Reply:
Having a meeting on this to just verify the language/compiler side. But given the value here, I expect this will all be accepted. We will let you know once the decision is made.